src/widget/wbasewidget.cpp

```diff
@@ -139,63 +139,72 @@ void WBaseWidget::setControlParameterRightUp(double v) {
 }
 
 void WBaseWidget::updateTooltip() {
-    // If we are in developer mode, update the tooltip.
+    // Remove leading and trailing whitespace/line breaks from the tooltip.
+    QString base = baseTooltip().trimmed();
+
+    // If we are in developer mode, expand the tooltip.
     if (CmdlineArgs::Instance().getDeveloper()) {
         QStringList debug;
         fillDebugTooltip(&debug);
 
-        QString base = baseTooltip();
         if (!base.isEmpty()) {
-            debug.append(QString("Tooltip: \"%1\"").arg(base));
+            debug.append(QStringLiteral("Tooltip: \"%1\"").arg(base));
         }
-        m_pWidget->setToolTip(debug.join("\n"));
+        m_pWidget->setToolTip(debug.join(QChar('\n')));
+    } else {
+        m_pWidget->setToolTip(base);
     }
 }
 
-template <>
+template<>
 QString toDebugString(const QSizePolicy::Policy& policy) {
     switch (policy) {
-        case QSizePolicy::Fixed:
-            return "Fixed";
-        case QSizePolicy::Minimum:
-            return "Minimum";
-        case QSizePolicy::Maximum:
-            return "Maximum";
-        case QSizePolicy::Preferred:
-            return "Preferred";
-        case QSizePolicy::Expanding:
-            return "Expanding";
-        case QSizePolicy::MinimumExpanding:
-            return "MinimumExpanding";
-        case QSizePolicy::Ignored:
-            return "Ignored";
-        default:
-            break;
-    }
-    return QString::number(static_cast<int>(policy));
+    case QSizePolicy::Fixed:
+        return QStringLiteral("Fixed");
+    case QSizePolicy::Minimum:
+        return QStringLiteral("Minimum");
+    case QSizePolicy::Maximum:
+        return QStringLiteral("Maximum");
+    case QSizePolicy::Preferred:
+        return QStringLiteral("Preferred");
+    case QSizePolicy::Expanding:
+        return QStringLiteral("Expanding");
+    case QSizePolicy::MinimumExpanding:
+        return QStringLiteral("MinimumExpanding");
+    case QSizePolicy::Ignored:
+        return QStringLiteral("Ignored");
+    default:
+        return QString::number(static_cast<int>(policy));
+    }
```

Comment:
this does the trick, just without the boilerplate
`return QVariant::fromValue(policy).toString();`

Reply:
naa,  meant as replacement for the entire switch.

Reply:
I did test it earlier, works beautifully

Reply:
Done!